Lib/logging/handlers.py

```diff
@@ -187,14 +187,17 @@ def shouldRollover(self, record):
         Basically, see if the supplied record would cause the file to exceed
         the size limit we have.
         """
+        # See bpo-45401: Never rollover anything other than regular files
+        if not os.path.isfile(self.baseFilename):
```

Comment:
`isfile()` returns False if the file does not exist. Please add tests for `delay=True`.